plugins/uefi-capsule/fu-plugin-uefi-capsule.c

```diff
@@ -23,13 +23,102 @@ struct FuPluginData {
 	FuUefiBgrt *bgrt;
 	FuVolume *esp;
 	FuBackend *backend;
+	GFile *fwupd_efi_file;
+	GFileMonitor *fwupd_efi_monitor;
 };
 
+static gboolean
+fu_plugin_uefi_capsule_fwupd_efi_parse(FuPlugin *plugin, GError **error)
+{
+	FuPluginData *data = fu_plugin_get_data(plugin);
+	FuContext *ctx = fu_plugin_get_context(plugin);
+	const guint8 needle[] = "f\0w\0u\0p\0d\0-\0e\0f\0i\0 \0v\0e\0r\0s\0i\0o\0n\0 ";
+	gsize bufsz = 0;
+	gsize offset = 0;
+	guint16 version_tmp[16] = {0x0};
+	g_autofree gchar *buf = NULL;
+	g_autofree gchar *version = NULL;
+
+	/* find the UTF-16 version string */
+	if (!g_file_load_contents(data->fwupd_efi_file, NULL, &buf, &bufsz, NULL, error))
+		return FALSE;
+	if (!fu_memmem_safe((const guint8 *)buf, bufsz, needle, sizeof(needle), &offset, error)) {
+		g_autofree gchar *fn = g_file_get_path(data->fwupd_efi_file);
+		g_prefix_error(error, "searching %s: ", fn);
+		return FALSE;
+	}
+
+	/* align */
+	if (!fu_memcpy_safe((guint8 *)version_tmp,
+			    sizeof(version_tmp),
+			    0x0, /* dst */
+			    (const guint8 *)buf,
+			    bufsz,
+			    offset + sizeof(needle),
+			    sizeof(version_tmp) - sizeof(guint16),
+			    error))
+		return FALSE;
+
+	/* convert to UTF-8 */
+	version = g_utf16_to_utf8(version_tmp, -1, NULL, NULL, error);
+	if (version == NULL) {
+		g_autofree gchar *fn = g_file_get_path(data->fwupd_efi_file);
+		g_prefix_error(error, "converting %s: ", fn);
+		return FALSE;
+	}
+
+	/* success */
+	fu_context_add_runtime_version(ctx, "org.freedesktop.fwupd-efi", version);
+	return TRUE;
+}
+
+static void
+fu_plugin_uefi_capsule_fwupd_efi_changed_cb(GFileMonitor *monitor,
+					    GFile *file,
+					    GFile *other_file,
+					    GFileMonitorEvent event_type,
+					    gpointer user_data)
+{
+	FuPlugin *plugin = FU_PLUGIN(user_data);
+	FuContext *ctx = fu_plugin_get_context(plugin);
+	g_autoptr(GError) error_local = NULL;
+
+	if (!fu_plugin_uefi_capsule_fwupd_efi_parse(plugin, &error_local)) {
+		fu_context_add_runtime_version(ctx, "org.freedesktop.fwupd-efi", NULL);
```

Comment:
I think if you found the file but not the string you can make an assumption on the version can't you?

Reply:
I don't think so; or do you mean something like fwupd 1.3.whatever it was

Reply:
> I don't think so; or do you mean something like fwupd 1.3.whatever it was

Yeah that's what I mean. Set it to last release of fwupd-efi we have before this fix goes in.
It's a reasonable assumption and the most "common" case right now.

Reply:
> I think i'd rather have the number of firmwares depending on fwupd-efi to be as small as possible....

If that system is preproduction maybe it can be 0...

Reply:
> Set it to last release of fwupd-efi we have before this fix goes in

Surely, it should be the first release of fwupd-efi? we don't know it's not 1.1, 1.2. etc.

> If that system is preproduction

😬

Reply:
> > Set it to last release of fwupd-efi we have before this fix goes in
> 
> Surely, it should be the first release of fwupd-efi? we don't know it's not 1.1, 1.2. etc.
> 

Yeah

Reply:
Done.